spec/factories/ext_management_system.rb

```diff
@@ -4,7 +4,7 @@
     sequence(:hostname)  { |n| "ems-#{seq_padded_for_sorting(n)}" }
     sequence(:ipaddress) { |n| ip_from_seq(n) }
     guid                 { SecureRandom.uuid }
-    zone                 { Zone.first || FactoryGirl.create(:zone) }
+    zone                 { Zone.visible.first || FactoryGirl.create(:zone) }
```

Comment:
I wonder if the intention here was to get the default zone?  Maybe having a `Zone.default` would help?

There are a few other places that do `Zone.first` ... and one of them isn't in specs :scream: 
https://github.com/ManageIQ/manageiq/blob/master/app/models/container_deployment.rb#L60 <- this looks like an issue
The rest are just specs:
https://github.com/ManageIQ/manageiq/blob/master/spec/tools/environment_builders/openstack/helper_methods.rb
https://github.com/ManageIQ/manageiq/blob/master/spec/factories/provider.rb
https://github.com/ManageIQ/manageiq/blob/master/spec/models/metric/statistic_spec.rb

Reply:
@agrare it seems that answer is "sometimes it was intention, sometimes not". 
What do you mean by `Zone.default` -> `Zone.default_zone`? It doesn't work for all specs.
I fixed non-spec file, but added `visible` to spec files, which seems to be working (all other specs suppose they have visible zone)

Reply:
You might need to add `:name => "default"` to this `|| FactoryGirl.create(:zone)` call.  I'd prefer `Zone.default_zone` over `Zone.visible.first` in these specs.

Reply:
@agrare as I wrote, sometimes these definitions are not supposed to link to default_zone, but to some zone created somewhere before. 
For example `/spec/models/job_proxy_dispatcher_embedded_scan_spec.rb` doesn't work with default_zone. It expects some "Zone 1".
So the `Zone.visible.first` is necessary, because it doesn't change init data.

Reply:
I believe this last comment has been resolved by #18118